lib/chef/knife/windows_cert_generate.rb

```diff
@@ -75,7 +76,7 @@ def prompt_for_passphrase
       end
 
       def generate_certificate rsa_key
-        @hostname = "*"
+        @hostname = Socket.gethostname
```

Comment:
@btm 
Socket.gethostname will always pickup hostname. 
we made this changes for following use case-
1. User will run the windows cert generate command on remote server to create ssl cert
2. If --hostname is not specified then by default it will take server 'hostname' (i.e retrived  by using Socket.gethostname).
But now we realize generated cert is not working with hostname, it needs server's 'fqdn'.
And to find out 'fqdn' we can take help of reverse dns lookup. but is it right way to get fqdn? As i think dns reverse lookup is seems quite tricky!

or we should just make this fqdn option manadatory?